### PR TITLE
Refacto MultistoreController into components

### LINKED ISSUE #37017: Refacto src/PrestaShopBundle/Controller/Admin/MultistoreController
This one is a particular case, the `MultistoreController` is not a regular controller attached to some routing it is only used as a component and is rendered in the twig layouts thanks to `MultistoreComponentsExtension`

This is a bad pattern it should be replaced by a Twig Component instead, this has already been done for the regular multishop component via the `PrestaShopBundle\Twig\Component\MultistoreHeader` but it was not done the the specific multishop header for product page which is still handled by `MultistoreController::productHeader`

So this refacto actually implies:
- create a `ProductMultistoreHeader` similar to the `MultistoreHeader` except it will contain the specific logic from `MultistoreController::productHeader`
- replace the usages in twig of `{{ multistoreProductHeader(productId) }}` by `{{ component('ProductMultistoreHeader', {productId: productId}) }}`
- remove `MultistoreController` and `MultistoreComponentsExtension`

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x
| Description?      | Refacto src/PrestaShopBundle/Controller/Admin/MultistoreController because it was based on the old framework base controller, but mostly because it should be a component, there fore it was all refactored into Twig component for product header, and a dedicated service to render configuration dropdown This we can remove the initial controller completely
| Type?             | refacto
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | CI green and UI tests green
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/11570055447
| Fixed issue or discussion?     | Fixes #37017
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `branch` should be `develop` or `8.2.x`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!

(Note: this is an automated message, but answering it will reach a real human)

**Comment by ps-jarvis** [comment]
This pull request seems to contain new translation strings. I have summarized them below to ease up review:

- `Admin.Actions`
    - [ ] `View my store`

(Note: this is an automated message, but answering it will reach a real human)

File: src/PrestaShopBundle/Service/Form/MultistoreConfigurationDropdownRenderer.php
Comment (jolelievre) [inline src/PrestaShopBundle/Service/Form/MultistoreConfigurationDropdownRenderer.php:1]: This whole code was copied from the initial MultiStoreController, this PR doesn't intend improving it (except the injected services based on recent Context that are isofunctional)

**Review by FabienPapet — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]
nice !

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
